### ISSUE TITLE
AssertionError during the validation phase

### ISSUE THREAD

**Issue by akhanss** [yhenon/pytorch-retinanet#35]
Thanks for the great repo. I am facing the following issue when train a model during the validation phase.

  File "train.py", line 156, in main
    mAP = csv_eval.evaluate(dataset_val, retinanet) 
  File "/home/a_khanss/pytorch-retinanet/csv_eval.py", line 187, in evaluate
    all_detections     = _get_detections(generator, retinanet, score_threshold=score_threshold, max_detections=max_det
  File "/home/a_khanss/pytorch-retinanet/csv_eval.py", line 36, in _get_detections
    scores, labels, boxes = retinanet(data['img'].permute(2, 0, 1).cuda().float().unsqueeze(dim=0))
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/modules/module.py", line 477, in
 __call__
    result = self.forward(*input, **kwargs)
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line
 124, in forward
    return self.gather(outputs, self.output_device)
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line
 136, in gather
    return gather(outputs, output_device, dim=self.dim)
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", lin
e 67, in gather
   return gather_map(outputs)
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", lin
e 62, in gather_map
    return type(out)(map(gather_map, zip(*outputs)))
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/scatter_gather.py", lin
e 54, in gather_map
   return Gather.apply(target_device, dim, *outputs)
  File "/home/a_khanss/anaconda3/envs/pytorch-tf/lib/python3.6/site-packages/torch/nn/parallel/_functions.py", line 52
, in forward
    assert all(map(lambda i: i.is_cuda, inputs))
AssertionError

**Comment by zbxzc35** [comment]
I got this error too

**Comment by Kwongrf** [comment]
emmmmm , I got the same error

**Comment by Kwongrf** [comment]
My solution is moving this  `mAP = csv_eval.evaluate(dataset_val, retinanet)`  out of  `for iter_num, data in enumerate(dataloader_train):`  because calculating mAP should be done after one epoch.

**Comment by mahaling** [comment]
@Kwongrf the train.py already has that line of code out of the for loop.. I am still getting the same error.. Any idea on what else is wrong?

**Comment by ooooverflow** [comment]
The bug comes from 270 line in `model.py`   
`return [torch.zeros(0), torch.zeros(0), torch.zeros(0, 4)]`
Because it does not return the value in GPU device. Change this line like this:
`return [torch.zeros([1]).cuda(0), torch.zeros([1]).cuda(0), torch.zeros([1, 4]).cuda(0)]`
Do not forget to change the 0 to 1, otherwise there will be another bug about array bound
